### PR TITLE
tests: Clean up the existing tests.py

### PR DESCRIPTION
**Pull request by EchterAgo**
This cleans up the code of `tests.py` and makes it use the new functions from `utils.py`. This also makes the test sleep much less, ~~which is only still required before destroying the pool due to https://github.com/openzfsonwindows/openzfs/issues/282~~

### CODE REVIEW

**Comment by EchterAgo** [comment]
Draft because I still need to make the `driveletter` test work. With the current code it just creates new pools with the same drive letter over and over again.

**Comment by andrewc12** [comment]
ZFS used to not reuse drive letters.

So the test was meant to try to repeatedly create and destroy pools one after the other so that if it ran out of drive letters it would fail.

It looks like you might be trying to assign drive letters yourself? From my brief glances.

**Comment by EchterAgo** [comment]
> ZFS used to not reuse drive letters.
> 
> So the test was meant to try to repeatedly create and destroy pools one after the other so that if it ran out of drive letters it would fail.
> 
> It looks like you might be trying to assign drive letters yourself? From my brief glances.

Yes, `zfs_create` does assign a driveletter by default. I want to add some option to not assign a driveletter.

**Comment by EchterAgo** [comment]
I rebased this on top of #288 so maybe this finally passes now?

**Comment by EchterAgo** [comment]
Added another commit that removes `Start-Sleep` from test1

**Comment by andrewc12** [comment]
I only re-enabled that test because pytest 5 was crashing

I will straight up disable it after pytest5 passes

**Comment by EchterAgo** [comment]
@andrewc12 ok, removed.

File: contrib/windows/tests/tests.py [hunk 1]
```diff
@@ -1,413 +1,151 @@
...
-        #
-        #
-        # ret = runWithPrint(["zpool", "export", "-a"])
-        # if ret.returncode != 0:
-        #     print("FAIL")
-        # time.sleep(10)
-        #
-        # runWithPrint(["zpool", "destroy", "-f", "testsn02"])
-        # time.sleep(10)
-        # postTest()
```
Comment (andrewc12) [inline contrib/windows/tests/tests.py:340]: Can you recreate this and comment it out?

Comment (EchterAgo) [reply contrib/windows/tests/tests.py:340]: It crashes

Comment (EchterAgo) [reply contrib/windows/tests/tests.py:340]: image

Comment (andrewc12) [reply contrib/windows/tests/tests.py:340]: Means you've done it correctly. :)

Comment (EchterAgo) [reply contrib/windows/tests/tests.py:340]: See #289

Comment (andrewc12) [reply contrib/windows/tests/tests.py:340]: I love you

Comment (EchterAgo) [reply contrib/windows/tests/tests.py]: Thanks! I re-added the hanging test but left it commented out.

**Comment by EchterAgo** [comment]
Note that I still need to fix the drive letter test so it doesn't explicitly set a drive letter. I want to be able to detect what drive letter / mountpoint the newly created pool was assigned in `create_zpool`.

**Comment by EchterAgo** [comment]
I just got this failure, not sure what happened:

```
2023-10-14 11:52:54,832 - tests.tests - INFO - ============================================================
2023-10-14 11:52:54,832 - tests.tests - INFO - Running test: snapshot no hang
2023-10-14 11:52:55,339 - tests.tests - INFO - Drive letters after testsn01 pool create: {'testsn01': WindowsPath('E:/')}
zunmount(testsn01,E:/) running
zunmount(testsn01,E:/) returns 0
Traceback (most recent call last):
  File "Y:\openzfs\contrib\windows\tests\tests.py", line 164, in <module>
    main()
  File "Y:\openzfs\contrib\windows\tests\tests.py", line 100, in main
    allocate_file(pool.mount_path / "test01.file", 1 * Size.KIB)
  File "Y:\openzfs\contrib\windows\tests\utils.py", line 332, in allocate_file
    with open(path, "wb") as f:
         ^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\test01.file'
```

**Comment by EchterAgo** [comment]
Maybe we should also have a test to see if the drive letter is available again after unmount.

**Comment by EchterAgo** [comment]
We should also probably test if the pool is actually mounted where we expect it on each test.

**Comment by EchterAgo** [comment]
I added the checks in `zpool_create`.

**Comment by andrewc12** [comment]
> Maybe we should also have a test to see if the drive letter is available again after unmount.

> We should also probably test if the pool is actually mounted where we expect it on each test.

Yeah these are both good ideas

**Comment by EchterAgo** [comment]
FYI, I let this test loop for the last two days and only encountered the drive letter thing *once*. I still intend to fix it.

**Pull request merged**